http/param/param.go

```diff
@@ -217,6 +244,23 @@ func (p Parser) parsePathParam(r *http.Request, paramName string, v reflect.Valu
 	return nil
 }
 
+func (p Parser) parseFormParam(r *http.Request, paramName string, v reflect.Value) error {
+	if r.Method != http.MethodPost && r.Method != http.MethodPut && r.Method != http.MethodPatch {
+		return fmt.Errorf("struct's field was tagged for parsing the form parameter (%s) but request method is not POST, PUT or PATCH", paramName)
+	}
+	if err := r.ParseForm(); err != nil {
```

Comment:
(thought, but no change needed in the end)

initially, I thought it would be nice (or even needed, as request body likely cannot be read multiple times) to parse body only once (only if there is at least one field tagged as form parameter), and not for each form parameter. But looks like both `ParseForm` and `ParseMultipartForm` already do both "check this is only called once per request (idempotent)" and "save parse result elsewhere (req.PostForm and req.Form)", so probably no change needed here.